File: 300Plan/_002SelectionSort.py
arr = [2, 100, 3, 5, 0]

print(*arr)

for i in range(len(arr)):
    min_index = i
    for j in range(i + 1, len(arr)):
        if arr[j] < arr[min_index]:
            min_index = j

    if i is not min_index:
        arr[i], arr[min_index] = arr[min_index], arr[i]

print(*arr)

File: 300Plan/_002SelectionSort.java
import java.util.Arrays;

public class _002SelectionSort {
    public static void main(String[] args) {
        _002SelectionSort test = new _002SelectionSort();
        test.selectionSort(new int[]{2,100,3,5,0});
    }
    private void selectionSort(int[] arr) {
        if (arr == null || arr.length == 0) {
            return;
        }
        Arrays.stream(arr).forEach(ele -> System.out.print(ele + " "));
        int len = arr.length;
        for (int i = 0; i < len - 1; i++) {
            int min_index = i;
            for (int j = i + 1; j < len; j++) {
                if (arr[j] < arr[min_index]) { // find the minimum
                    min_index = j;
                }
            }
            //Swap the found minimum element with the first element
            if (i != min_index) {
                arr[i] = arr[i] ^ arr[min_index];
                arr[min_index] = arr[i] ^ arr[min_index];
                arr[i] = arr[i] ^ arr[min_index];
            }
        }
        System.out.println();
        Arrays.stream(arr).forEach(ele -> System.out.print(ele + " "));
    }
}
